### PR TITLE
[ODS-5665] Add ability to query organizations by an identification code

### PR DESCRIPTION
**Pull request by mjaksn**

### CODE REVIEW

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Providers/ResourceIdentificaitonCodeProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            bool result = _provider.TryGetIdentificationCodeProperties(
+                resource, out List<ResourceProperty> identificationCodeProperties);
+
+            //Assert
+            result.ShouldBeFalse();
+            identificationCodeProperties.ShouldBeNull();
+        }
+        
+        [Test]
+        public void TryFindIdentificationCodes_ShouldReturnTrueAndCorrectIdentificationCodeProperties_WhenABaseResourceHasIdentificationCode()
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Providers/ResourceIdentificaitonCodeProviderTests.cs:43]: Maybe rename this to remove "Base" from the final name segment? Course is not really a base class, and it fits with the test above which is testing `WhenResourceHasNoIdentificationCode` -- maybe just use `WhenResourceHasIdentificationCode`?

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Providers/ResourceIdentificaitonCodeProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "CourseIdentificationSystemDescriptor",
+                "AssigningOrganizationIdentificationCode",
+                "CourseCatalogURL",
+                "IdentificationCode"
+            });
+        }
+        
+        [Test]
+        public void TryFindIdentificationCodes_ShouldReturnTrueAndCorrectIdentificationCodeProperties_WhenADerivedResourceHasIdentificationCode()
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Providers/ResourceIdentificaitonCodeProviderTests.cs:71]: Definitely optional ... you could go with `WhenADerivedResourceHasInheritedIdentificationCode` here.

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite GetByExample Tests.postman_collection.json [hunk 3]
```diff
@@ -1950,6 +1952,206 @@
...
+										"type": "text/javascript",
+										"packages": {}
+									}
+								}
+							],
+							"request": {
+								"method": "GET",
+								"header": [],
+								"url": {
+									"raw": "{{ApiBaseUrl}}/data/v3/ed-fi/localEducationAgencies?educationOrganizationIdentificationSystemDescriptor=uri://ed-fi.org/EducationOrganizationIdentificationSystemDescriptor%23non-existatnt",
```
Comment (gmcelhanon) [inline Postman Test Suite/Ed-Fi ODS-API Integration Test Suite GetByExample Tests.postman_collection.json:2134]: Correct the spelling on `non-existent`...

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite GetByExample Tests.postman_collection.json [hunk 4]
```diff
@@ -1950,6 +1952,206 @@
 							"response": []
 						}
 					]
+				},
+				{
+					"name": "When filtering by IdentificationCode",
```
Comment (gmcelhanon) [inline Postman Test Suite/Ed-Fi ODS-API Integration Test Suite GetByExample Tests.postman_collection.json:1957]: I feel like these tests could really use some negative test data (data that would show up in the results if the implementation was incorrect).  For example, in a "Setup" folder create a new temporary School that has identification codes that match the value being searched but uses a different identification system. (Also add a "Teardown" folder that removes this School and use a separate subfolder in between to hold the test requests, as necessary).

So, for example, create a School with an identification code that uses `uri://ed-fi.org/EducationOrganizationIdentificationSystemDescriptor#Other` with a value of `255901001`. This school should also be returned when only searching by identification code `255901001` but should not be returned if both the identification system and code are used in the search.

image

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs [hunk 5]
```diff
@@ -376,15 +377,49 @@ private IList<Parameter> CreateGetByExampleParameters(OpenApiMetadataPathsResour
...
+                            .Contains(p.Parent.FullName)).ToList();
+                }
+                
+                IEnumerableExtensions.ForEach(
+                    queryableIdentificationCodeProperties, x =>
+                    {
+                        parameterList.Add(
+                            new Parameter
+                            {
+                                name = (x.DescriptorName ?? x.PropertyName).ToCamelCase(),
```
[the remark this answers is not in the document]
Comment (mjaksn) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs]: I was thinking of entities and concerned descriptor property names could be suffixed with `Id`. That's not an issue with resources, so I changed it to only use `PropertyName` in the latest commit.

**Review by gmcelhanon — APPROVED** [review]

**Pull request merged**